module/users/helper.js

```diff
@@ -902,6 +971,93 @@ module.exports = class UsersHelper {
   }
 
 
+  /**
+   * Find non-targeted joined program.
+   * @method
+   * @name getUserJoinedProgramDetailsWithPreviousProfiles
+   * @param {Object} requestedData - requested data
+   * @returns {Object} - non-targeted joined program details.
+   */
+
+  static getUserJoinedProgramDetailsWithPreviousProfiles( bodyData, pageNo, pageSize, searchText, userId ) {
+    return new Promise(async (resolve, reject) => {
+      try {
+        // step 1 -> get all targeted program's Id, to the user profile.
+        let queryData = await programsHelper.queryBasedOnRoleAndLocation(
+          bodyData
+        );
+        
+        if( !queryData.success ) {
+          return resolve(queryData);
+        }
+
+        // get all targeted program ids.
+        // we have to recall this function here because we need to get all program details. ie) pageNo := 1 && pageSize:= toatal targeted program count.
+        let allTargetedPrograms = await programsHelper.list(
+          pageNo,
+          pageSize,
+          "",
+          queryData.data,
+          ["_id"]
+        );
+        let targettedProgramIds = [];
+        let programUsersIds = [];
+      
+        if ( allTargetedPrograms.success && allTargetedPrograms.data && allTargetedPrograms.data.data.length > 0) {
```

Comment:
@VISHNUDAS-tunerlabs what is the difference between allTargetedPrograms.data and queryData.data

Reply:
@aks30 queryData.data contain query constructed based on user role and location. Based on this we will fetch all program ids targeted to user (allTargetedPrograms.data)